app/Models/Store/Order.php

```diff
@@ -340,11 +340,15 @@ public function cancel()
 
     public function delete()
     {
-        if ($this->isModifiable() === false) {
-            // in most cases this would return a null key because the lookup for the cart
-            // would return a new cart anyway?
-            throw new Exception("Delete not allowed on Order ({$this->getKey()}).");
-        }
+        DB::connection($this->connection)->transaction(function () {
```

Comment:
`$this->getConnection()->transaction(...)`  looks like the same thing.

Reply:
making a separate PR for the existing ones